jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class HiveUrlUtils {
+    public static final String HIVE_DEFAULT_DATABASE = "";
+    public static final int HIVE_DEFAULT_PORT = 10001;
+    public static final String URL_JDBC_PREFIX = "jdbc:";
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url verified url passed in by client, already of valid format possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        // Stripped out for the purpose of using URI
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath() will return: /<dbName>;<sessionConfs>, substring gets rid of the "/"
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+        // getQuery() will return the info after '?'
+        String hiveConfs = jdbcURI.getQuery();
+        // getFragmeng() will return the info after '#'
+        String hiveVars = jdbcURI.getFragment();
+
+        String[] params = dbAndSessionConfs.split(";");
+        // Incorrect database name provided
+        if (params.length != 0 && params[0].contains("=")) {
+            throw new InvalidURLException(
+                    String.format(
+                            "Error: %s \n" + "Database name must not contain '='. Empty if no dbName specified.",
```

Comment:
As discussed offline, let's make this error message a bit clearer if the user was intending to pass parameters. For example, add a usage message.